state/syncer/missingTrieNodesNotifier.go

```diff
@@ -0,0 +1,53 @@
+package syncer
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-go/common"
+)
+
+type missingTrieNodesNotifier struct {
+	handlers []common.StateSyncNotifierSubscriber
+	mutex    sync.RWMutex
+}
+
+// NewMissingTrieNodesNotifier creates a new missing trie nodes notifier
+func NewMissingTrieNodesNotifier() *missingTrieNodesNotifier {
+	return &missingTrieNodesNotifier{
+		handlers: make([]common.StateSyncNotifierSubscriber, 0),
+		mutex:    sync.RWMutex{},
+	}
+}
+
+// RegisterHandler registers a new handler for the missing trie nodes notifier
+func (mtnn *missingTrieNodesNotifier) RegisterHandler(handler common.StateSyncNotifierSubscriber) {
```

Comment:
could have returned the error and removed the log.Warn print. Easier to test and also prevents an invalid setup that is obvious only by parsing the logs.

Reply:
Refactored.